python/graphstorm/model/utils.py

```diff
@@ -1361,28 +1379,36 @@ def load_sparse_emb(target_sparse_emb, ntype_emb_path):
     world_size = get_world_size()
     num_files = len(os.listdir(ntype_emb_path))
     num_embs = target_sparse_emb.num_embeddings
-    # Suppose a sparse embedding is trained and saved using N trainers (e.g., GPUs).
-    # We are going to use K trainers/infers to load it.
-    # The code handles the following cases:
-    # 1. N == K
-    # 2. N > K, some trainers/infers need to load more than one files
-    # 3. N < K, some trainers/infers do not need to load any files
-    for i in range(math.ceil(num_files/world_size)):
-        file_idx = i * world_size + rank
-        if file_idx < num_files:
-            emb = th.load(os.path.join(ntype_emb_path,
-                                       f'sparse_emb_{pad_file_index(file_idx)}.pt'))
-
-            # Get the target idx range for sparse_emb_{rank}.pt
-            start, end = _get_sparse_emb_range(num_embs,
-                                                rank=file_idx,
-                                                world_size=num_files)
-            # write sparse_emb back in an iterative way
-            batch_size = 10240
-            idxs = th.split(th.arange(end - start), batch_size, dim=0)
-            for idx in idxs:
-                # TODO: dgl.distributed.DistEmbedding should allow some basic tensor ops
-                target_sparse_emb._tensor[start+idx] = emb[idx]
+
+    if isinstance(target_sparse_emb, WholeGraphSparseEmbedding):
+        # Using WholeGraph will load sparse emb in binary format, let's assume
+        # the sparse emb is saved by WholeGraphSparseEmbedding.save_to_file(), i.e.,
+        # the meta info remains the same.
+        # Example: wg_sparse_emb_part_1_of_2, wg_sparse_emb_part_2_of_2
+        target_sparse_emb.load_from_file(ntype_emb_path, "wg_sparse_emb", num_files)
```

Comment:
Do we allow save wholegraph sparse tensor into 8 parts with 8 GPUs but load them using only 4 GPUs?
If not, can we add some check here?

Reply:
As long as we use all procs to load all files, it should be fine, i.e., num_files does not need to be the same as num_training_procs.
For example, say we save WG sparse emb saved in 8 files; then later, we can load them with arbitrary procs as long as all wholegraph processes are participated.

Reply:
Another question, does it require each machine participating in the training has all the wholegraph files under ntype_emb_path?

Reply:
In WhoelGraphDistTensor.load_from_file, wg_optimizer is required if we are loading a trainable sparse embedding. But his is not handled here. Shall we pass in a wg_optimizer here or we will do this later when initializing the optimizer.

Reply:
>Another question, does it require each machine participating in the training has all the wholegraph files under ntype_emb_path?

Unfortunately, for now, yes.

Reply:
> wg_optimizer is required if we are loading a trainable sparse embedding.

I see. When loading a trainable embeddings, do we assume `wg_sparse_optimizer` is already available at: target_sparse_emb._optimizer? i.e., would `target_sparse_emb` be attached a optimizer via `init_optimizer()` function?
If so (target_sparse_emb._optimizer is not None), then I think we don't need to pass an additional optimizer to the function.

Reply:
By default optimizer is initialized before restoring model params.